packages/compiler-cli/ngcc/src/entry_point_finder/utils.ts

```diff
@@ -118,3 +94,76 @@ export function trackDuration<T = void>(task: () => T extends Promise<unknown>?
   log(duration);
   return result;
 }
+
+/**
+ * Remove paths that are contained by other paths.
+ *
+ * For example:
+ * Given `['a/b/c', 'a/b/x', 'a/b', 'd/e', 'd/f']` we will end up with `['a/b', 'd/e', 'd/f]`.
+ * (Note that we do not get `d` even though `d/e` and `d/f` share a base directory, since `d` is not
+ * one of the base paths.)
+ *
+ */
+function dedupePaths(paths: AbsoluteFsPath[]) {
```

Comment:
I would add an explicit return type here:

```suggestion
function dedupePaths(paths: AbsoluteFsPath[]): AbsoluteFsPath[] {
```